core/src/main/scala/org/apache/spark/broadcast/TorrentBroadcast.scala

```diff
@@ -186,7 +186,7 @@ private[spark] class TorrentBroadcast[T: ClassTag](obj: T, id: Long)
               }
               blocks(pid) = new ByteBufferBlockData(b, true)
             case None =>
-              throw new SparkException(s"Failed to get $pieceId of $broadcastId")
```

Comment:
Why not just catch `SparkException` outside?

Reply:
I think `BlockNotFoundException` is more clear, and will make it simple when checking the exception types.